File: frontend/src/features/roles/types/role.ts
export interface RoleRecord {
  id: number;
  name: string;
  code: string;
  description?: string;
  permissions: string[];
  is_system?: boolean;
}

export interface RoleFilters {
  search: string;
}

export interface RoleFormValues {
  name: string;
  code: string;
  description: string;
  permissions: string;
}

File: backend/apps/users/models/role.py
# backend/apps/users/models/role.py
from django.db import models


class Role(models.Model):
    name = models.CharField(max_length=100, unique=True)
    code = models.SlugField(max_length=100, unique=True)
    description = models.TextField(blank=True)
    permissions = models.JSONField(default=list, blank=True)
    is_system = models.BooleanField(default=False)
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        ordering = ["name"]

    def __str__(self) -> str:
        return self.name

    def has_permission(self, permission_code: str) -> bool:
        return permission_code in set(self.permissions or [])
